runtime_scan/pkg/provider/docker/client.go

```diff
@@ -0,0 +1,585 @@
+			}
+			return
+		}
+		assets <- asset
+	}
+}
+
+func (c Client) getContainers(ctx context.Context, assets chan models.AssetType, errs chan error, wg *sync.WaitGroup) {
+	defer wg.Done()
+
+	containers, err := c.dockerClient.ContainerList(ctx, types.ContainerListOptions{
+		All: true,
+	})
+	if err != nil {
+		errs <- provider.FatalError{
+			Err: fmt.Errorf("failed to get containers. Provider=%s: %w", models.Docker, err),
+		}
+		return
+	}
+
+	for _, container := range containers {
+		inspect, err := c.dockerClient.ContainerInspect(ctx, container.ID)
+		if err != nil {
+			errs <- provider.FatalError{
+				Err: fmt.Errorf("failed to get container. Provider=%s: %w", models.Docker, err),
+			}
+			return
+		}
+
+		created, err := time.Parse(time.RFC3339, inspect.Created)
+		if err != nil {
+			errs <- provider.FatalError{
+				Err: fmt.Errorf("failed to parse time. Provider=%s: %w", models.Docker, err),
+			}
+			return
+		}
+
+		imageInfo, err := c.getContainerImageInfoFromImage(ctx, container.ImageID)
+		if err != nil {
+			// TODO (paralta) If image not required this should not be fatal
+			errs <- provider.FatalError{
+				Err: err,
+			}
+			return
+		}
+
+		asset := models.AssetType{}
+		err = asset.FromContainerInfo(models.ContainerInfo{
+			ContainerName: utils.PointerTo(inspect.Name),
+			CreatedAt:     utils.PointerTo(created),
+			Id:            utils.PointerTo(container.ID),
+			Image:         utils.PointerTo(imageInfo),
+			Labels:        convertTags(container.Labels),
+			Location:      nil, // TODO (paralta) Clarify what is location
+			ObjectType:    "ContainerInfo",
+		})
+		if err != nil {
+			errs <- provider.FatalError{
+				Err: fmt.Errorf("failed to create AssetType from ContainerInfo. Provider=%s: %w", models.Docker, err),
+			}
+			return
+		}
+		assets <- asset
+	}
+}
+
+func (c Client) RunAssetScan(ctx context.Context, config *provider.ScanJobConfig) error {
+
+	scanName, err := getScanName(config)
+	if err != nil {
+		return provider.FatalError{Err: err}
+	}
+	resp, err := c.dockerClient.VolumeList(ctx, volume.ListOptions{
+		Filters: filters.NewArgs(filters.Arg("name", scanName)),
+	})
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to get volumes. Provider=%s: %w", models.Docker, err),
+		}
+	}
+	if len(resp.Volumes) == 0 {
+		_, err = c.dockerClient.VolumeCreate(ctx, volume.CreateOptions{
+			Name: scanName,
+		})
+		if err != nil {
+			return provider.FatalError{
+				Err: fmt.Errorf("failed to create volume. Provider=%s: %w", models.Docker, err),
+			}
+		}
+	}
+
+	readCloser, err := c.export(ctx, config)
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to export target. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	// Create an ephemeral container to populate volume with export output
+	_, err = c.dockerClient.ImagePull(ctx, "alpine", types.ImagePullOptions{})
```

Comment:
If we will be pulling VMClarity CLI tool, then we can also try using that image to perform bootstrapping rather than doing it with some other image. Maybe consider this at the end.